### DISCUSSION TITLE
How to test UesrNameToken

### DISCUSSION THREAD

**Discussion by fyspkid** [onvif/testspecs#52] [Q&A] [answered]
Hi,
1. I am wondering is the test case SECURITY-1-1-1-v14.12 USER TOKEN PROFILE is a mandertory test item for onvif device conformance test on Profile S?
2. The test case is always fail with the error message below
    STEP 1 - Check if credentials were defined
      StepStart: 2023-08-16T09:54:55.3806966Z
      This test cannot be performed without credentials supplied
    STEP FAILED
<img width="637" alt="usertoken2" src="https://github.com/onvif/testspecs/assets/15334934/0c0c83e2-a52d-4f4c-98a7-692d251d8b60">
<img width="659" alt="usertoken" src="https://github.com/onvif/testspecs/assets/15334934/b19b9c64-da50-4cc1-85f6-d242db8d72bd">

 Is there anything I missed in onvif device test tool (V23.06 rev.96) ?

3. Here are the test logs and some test messages, is there anything I need to modify in my code for example GetService or GetCapabilities features in device managment?
[UserToken.zip](https://github.com/onvif/testspecs/files/12358195/UserToken.zip)
<img width="761" alt="usertoken1" src="https://github.com/onvif/testspecs/assets/15334934/f36889f5-3575-462a-b47f-6dc522d7e7f9">

Thak you so much for your help.

**Comment by fyspkid** [comment]
Hi
Today we tried to get the soap header with WS UsernameToken and return soap fault on Device Management GetCapabilities  &  GetServices features as the test spec "ONVIF Device Feature Discovery Specification V21.12".
<img width="320" alt="Screenshot 2023-08-17 at 00 47 11" src="https://github.com/onvif/testspecs/assets/15334934/a24a15c2-7964-4c88-ab33-52f084c23a18">
But it seems the onvif device test tool only send requests without any authentication. 
We also double check with wireshark and didn't see any requests with WS-UsernameToken authentication.

How to make the device test tool send request with WS-UsernameToken authentication during testing SECURITY-1-1-1-v14.12 USER TOKEN PROFILE ?

Thank you so much

**Comment by madhurao68** [comment]
Hello,
Please upload log files to help us investigate. Thanks for your patience!

**Comment by fyspkid** [comment, marked as answer]
Hi madhurao68 
Thank you for your response.
By filling out the administrator username and password in the onvif device tool's discovery tab, we were able to solve the issue on our own.
I will close this Q&A.

By the way, the log file zip is in the upper question field.
I really appreciate your time and response.
Have a nice day.
